.github/workflows/auto-update-gemfiles.yml

```diff
@@ -0,0 +1,169 @@
+name: Auto-update Gemfiles
+
+# This action cannot be skipped altogether because it is a mandatory status check.
+# Instead we conditionally skip it at job level, instead of workflow level.
+on:
+  schedule:
+    - cron: '0 0 * * 0' # Every Sunday at midnight
+  workflow_dispatch:
+
+# Ensure obsolete job is cancelled if another commit is pushed to the same branch.
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
```

Comment:
how does `||` operator work here? can `github.event.pull_request.number` be falsey?

Reply:
The workflow cannot be triggered by `pull_request`

Reply:
yeah, the workflow cannot triggered by pull request, the code seems to be copy & pasted without changes.